src/apb/engine.py

```diff
@@ -111,47 +140,74 @@ def make_friendly(blob):
     return flines
 
 
-def touch(fname):
+def touch(fname, force):
     if os.path.exists(fname):
         os.utime(fname, None)
+        if force:
+            os.remove(fname)
+            open(fname, 'a').close()
     else:
         open(fname, 'a').close()
 
 
-def init_actions(project_path, provider, ansible_dir_exists):
-    actions_path = os.path.join(project_path, ACTIONS_DIR)
-    src_file = 'shipit-%s.yml' % provider
-    provision_src_path = os.path.join(project_path, 'ansible', src_file)
-    provision_dest_path = os.path.join(actions_path, 'provision.yaml')
-
-    if not os.path.exists(actions_path):
-        os.makedirs(actions_path)
-
-    if ansible_dir_exists:
-        # Only write over provision.yml if the file doesn't already exist
-        if not os.path.exists(provision_dest_path):
-            copyfile(provision_src_path, provision_dest_path)
-    else:
-        print('NOTE: No ansible dir found at project root.' +
-              'Assuming manual authoring.')
-
-
-def init_dockerfile(spec_path, dockerfile_path, ansible_dir_exists):
+def update_dockerfile(spec_path, dockerfile_path):
     # TODO: Defensively confirm the strings are encoded
     # the way the code expects
     blob = base64.b64encode(load_spec_str(spec_path))
     dockerfile_out = insert_encoded_spec(
-        load_dockerfile(ansible_dir_exists), make_friendly(blob)
+        load_dockerfile(dockerfile_path), make_friendly(blob)
     )
 
-    write_dockerfile(dockerfile_out, dockerfile_path)
+    write_dockerfile(dockerfile_out, dockerfile_path, False)
     print('Finished writing dockerfile.')
 
 
+def cmdrun_init(**kwargs):
+    print("Initializing current directory for an APB")
+    current_path = kwargs['base_path']
+    apb_name = kwargs['name']
+    project = os.path.join(current_path, apb_name)
+
+    if os.path.exists(project):
+        if not kwargs['force']:
+            raise Exception('ERROR: Project directory: [%s] found and force option not specified' % project)
+        shutil.rmtree(project)
+
+    os.mkdir(project)
+
+    spec_path = os.path.join(project, SPEC_FILE)
+    playbooks_path = os.path.join(project, PLAYBOOKS_DIR)
+    roles_path = os.path.join(project, ROLES_DIR)
+    dockerfile_path = os.path.join(os.path.join(project, DOCKERFILE))
+
+    if os.path.exists(spec_path) and not kwargs['force']:
+        raise Exception('ERROR: Spec file: [%s] found and force option not specified' % spec_path)
```

Comment:
Is this ever a real case? `apb init` seems to fail if a project exists without --force. I guess it's ok defensively. But I don't think the previous project dir check will let it get this far.